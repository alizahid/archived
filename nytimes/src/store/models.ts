import { Action, Thunk, action, thunk } from 'easy-peasy'

import { api } from '../lib'

export interface Article {
  abstract: string
  byline: string
  id: number
  image: string
  published: any
  thumb: string
  title: string
  url: string
  views: number
}

export interface ArticlesModel {
  interval: number
  loading: boolean

  articles: Article[]

  fetchArticles: Thunk<ArticlesModel, number>

  setInterval: Action<ArticlesModel, number>
  setLoading: Action<ArticlesModel, boolean>
  setArticles: Action<ArticlesModel, Article[]>
}

const articles: ArticlesModel = {
  interval: 1,
  loading: false,

  articles: [],

  fetchArticles: thunk(async (actions, payload) => {
    actions.setLoading(true)

    const articles = await api.fetchArticles(payload)

    actions.setArticles(articles)

    actions.setLoading(false)
  }),

  setInterval: action((state, payload) => {
    state.interval = payload
  }),
  setLoading: action((state, payload) => {
    state.loading = payload
  }),
  setArticles: action((state, payload) => {
    state.articles = payload
  })
}

export interface StoreModel {
  articles: ArticlesModel
}

const model: StoreModel = {
  articles
}

export default model
