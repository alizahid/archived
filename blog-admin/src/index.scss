// fonts

@import url('https://rsms.me/inter/inter.css');
@import url('https://fonts.googleapis.com/css?family=IBM+Plex+Mono:400,700&display=swap');

// reset

* {
  border-radius: 0;
  box-sizing: border-box;
  font-weight: normal;
  margin: 0;
  outline: none;
  padding: 0;
  user-select: none;
  -webkit-tap-highlight-color: transparent;
}

// global

a {
  color: #ff2d55;
  cursor: pointer;
  text-decoration: none;
  transition: 0.2s;

  &:hover {
    filter: brightness(1.2);
  }
}

body {
  background: #fff;
  color: #000;
  margin: 0;
  font: normal 1em/1 'Inter var', -apple-system, BlinkMacSystemFont, 'Segoe UI',
    'Roboto', 'Oxygen', 'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans',
    'Helvetica Neue', sans-serif;
  overflow: hidden;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font: normal 1em/1 'IBM Plex Mono', source-code-pro, Menlo, Monaco, Consolas,
    'Courier New', monospace;
  line-height: 1.6;
}

form {
  label {
    display: block;
    margin: 1em 0;

    span {
      display: block;
      font-weight: 500;
      margin-bottom: 1em;
    }
  }

  button,
  input,
  textarea {
    appearance: none;
    background: #fff;
    border: 1px solid #ecf0f1;
    border-radius: 0.25em;
    font: inherit;
    padding: 1em;
    transition: 0.3s;
  }

  button {
    background: #ecf0f1;
    border: none;
    cursor: pointer;
    line-height: normal;
    padding: 1em 2em;

    &:hover {
      background: #bdc3c7;
    }
  }

  input,
  textarea {
    width: 100%;

    &:focus {
      border-color: #bdc3c7;
    }
  }

  textarea {
    line-height: 1.4;
    min-height: 10em;
  }
}

h1,
h2,
h3 {
  font-weight: 600;
  margin: 1em 0;
}

h1 {
  font-size: 4em;
}

h2 {
  font-size: 3em;
  margin-bottom: 0.5em;
}

h3 {
  font-size: 2em;
  margin-bottom: 0.5em;
}

hr {
  border: none;
  height: 1px;
  margin: 2em 0;
  position: relative;
}

hr:before {
  background: linear-gradient(
    to left,
    rgba(0, 0, 0, 0.01),
    rgba(0, 0, 0, 0.05),
    rgba(0, 0, 0, 0.01)
  );
  content: '';
  height: 100%;
  left: 0;
  position: absolute;
  top: 0;
  width: 100%;
}

img {
  max-width: 100%;
  vertical-align: top;
}

ol,
ul {
  margin: 1em 0;
}

ol li,
ul li {
  line-height: 1.6;
  list-style: square;
  margin: 1em 0 1em 2em;
}

p {
  margin: 1em 0;
  line-height: 1.6;
}

// layout

#root {
  display: flex;
  flex-direction: column;
  height: 100vh;
}
