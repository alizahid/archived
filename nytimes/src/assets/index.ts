export { default as icon_date } from './icons8-calendar.png'
export { default as icon_views } from './icons8-visible.png'
