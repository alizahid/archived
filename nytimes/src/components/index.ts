export { default as Full } from './full'
export { default as Loading } from './loading'
export { default as NotFound } from './not-found'
export { default as Periods } from './periods'
export { default as Preview } from './preview'
export { default as Separator } from './separator'
