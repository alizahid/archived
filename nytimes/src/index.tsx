import React, { FunctionComponent } from 'react'
import { createAppContainer, createStackNavigator } from 'react-navigation'
import { StoreProvider } from 'easy-peasy'

import { Article, Home } from './scenes'
import { store } from './store'

const Navigator = createStackNavigator({
  Home,
  Article
})

const NYTimes: FunctionComponent = () => {
  const Container = createAppContainer(Navigator)

  return (
    <StoreProvider store={store}>
      <Container />
    </StoreProvider>
  )
}

export default NYTimes
