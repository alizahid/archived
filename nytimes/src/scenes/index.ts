export { default as Article } from './article'
export { default as Home } from './home'
