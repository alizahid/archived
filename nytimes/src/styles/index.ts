import { iOSColors } from 'react-native-typography'

export const layout = {
  icon: 20,
  image: 50,
  margin: 20,
  padding: 10,
  radius: 5
}

export const shadow = {
  elevation: 5,
  shadowColor: iOSColors.gray,
  shadowOffset: {
    height: 0,
    width: 0
  },
  shadowOpacity: 0.25,
  shadowRadius: 5
}
