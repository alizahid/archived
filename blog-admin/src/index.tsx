import './index.scss'

import React, { FunctionComponent } from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter, Route } from 'react-router-dom'

import { Header } from './components'
import { Auth, Posts, Projects } from './scenes'
import * as serviceWorker from './serviceWorker'

const Blog: FunctionComponent = () => (
  <BrowserRouter>
    <Header />
    <Route path="/" exact component={Auth} />
    <Route path="/posts" component={Posts} />
    <Route path="/projects" component={Projects} />
  </BrowserRouter>
)

ReactDOM.render(<Blog />, document.getElementById('root'))

serviceWorker.unregister()
