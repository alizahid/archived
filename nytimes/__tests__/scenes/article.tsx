import 'react-native'
import React from 'react'
import { create } from 'react-test-renderer'

import { StoreProvider } from 'easy-peasy'

import Article from '../../src/scenes/article'
import { store } from '../../src/store'

import { data } from '../fixtures'

beforeAll(async () => {
  fetch.mockResponseOnce(JSON.stringify(data))

  await store.dispatch.articles.fetchArticles(1)
})

it('renders', async () => {
  const navigation: any = {
    getParam: () => 100000006515727
  }

  const tree = create(
    <StoreProvider store={store}>
      <Article navigation={navigation} />
    </StoreProvider>
  )

  expect(tree).toMatchSnapshot()
})

it('not found', async () => {
  const navigation: any = {
    getParam: () => 1
  }

  const tree = create(
    <StoreProvider store={store}>
      <Article navigation={navigation} />
    </StoreProvider>
  )

  expect(tree).toMatchSnapshot()
})
