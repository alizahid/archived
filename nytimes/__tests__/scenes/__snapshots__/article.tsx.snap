// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`not found 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#000000",
        "fontFamily": "System",
        "fontSize": 20,
        "fontWeight": "600",
        "letterSpacing": 0.75,
        "lineHeight": 25,
      }
    }
  >
    Not found
  </Text>
</View>
`;

exports[`renders 1`] = `
<SafeAreaView
  forceInset={
    Object {
      "top": "never",
    }
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View>
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "elevation": 5,
              "shadowColor": "#8E8E93",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 5,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://static01.nyt.com/images/2019/05/28/us/politics/00dc-ufo2/merlin_98047735_3653fd43-cc8d-4ece-8947-bc0653b1d92d-jumbo.jpg",
              }
            }
            style={
              Object {
                "backgroundColor": "#000000",
                "height": 562.5,
                "width": 750,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 34,
                "fontWeight": "700",
                "letterSpacing": 0.41,
                "lineHeight": 41,
              }
            }
          >
            ‘Wow, What Is That?’ Navy Pilots Report Unexplained Flying Objects
          </Text>
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#8E8E93",
                "fontFamily": "System",
                "fontSize": 13,
                "fontWeight": "400",
                "letterSpacing": -0.078,
                "lineHeight": 18,
                "marginTop": 10,
              }
            }
          >
            By Helene Cooper Ralph Blumenthal And Leslie Kean
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "marginLeft": -20,
                "marginTop": 10,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "marginLeft": -20,
                    "marginTop": 10,
                  },
                  Object {
                    "marginLeft": 20,
                    "marginTop": 0,
                  },
                ]
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/assets/icons8-calendar.png",
                  }
                }
                style={
                  Object {
                    "height": 20,
                    "width": 20,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#8E8E93",
                    "fontFamily": "System",
                    "fontSize": 13,
                    "fontWeight": "400",
                    "letterSpacing": -0.078,
                    "lineHeight": 18,
                    "marginLeft": 10,
                  }
                }
              >
                5/26/2019
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "marginLeft": -20,
                    "marginTop": 10,
                  },
                  Object {
                    "marginLeft": 20,
                    "marginTop": 0,
                  },
                ]
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/assets/icons8-visible.png",
                  }
                }
                style={
                  Object {
                    "height": 20,
                    "width": 20,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#8E8E93",
                    "fontFamily": "System",
                    "fontSize": 13,
                    "fontWeight": "400",
                    "letterSpacing": -0.078,
                    "lineHeight": 18,
                    "marginLeft": 10,
                  }
                }
              >
                1
              </Text>
            </View>
          </View>
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 17,
                "fontWeight": "400",
                "letterSpacing": -0.408,
                "lineHeight": 22,
                "marginTop": 20,
              }
            }
          >
            No one at the Pentagon is saying that the objects are extraterrestrial, but the Navy has issued new classified guidance for reporting unexplained aerial phenomena.
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</SafeAreaView>
`;
